### PR TITLE
feat(Datatable): table header

### PR DESCRIPTION
**Pull request by Ratheeshrajan**
Contributes to #1931 

WKC data table does not include Carbon for IBM Products PAL table headers.

#### What did you change?

- `title` and `description` props(supplied via `datgridState`) added in `<TableContainer />` component. 
- `<DatagridToolbar />` moved inside `<TableContainer />`
- Fix overflow scroll of table container

#### How did you test and verify your work?
Storybook

### CODE REVIEW

**Comment by Ratheeshrajan** [comment]
@matthewgallo Thanks for spotting the issue in the left panel. I have fixed that one in the last commit and now it looks like the following screenshot. 
Screenshot 2022-05-13 at 3 30 40 PM

I have moved the left panel inside `TableContainer` to fix that issue.

**Comment by Ratheeshrajan** [comment]
> Hey @Ratheeshrajan, the fix for the left panel story/placement of the toolbar looks great. But my only reservation now is that there seems to be double scrollbars on some cases which is something we should fix.
> 
> This screenshot is from the `Batch actions` story: image

@matthewgallo fixed the double scrollbar issue. Please have a look.

File: packages/cloud-cognitive/src/components/Datagrid/_storybook-styles.scss
Comment (elycheea) [inline packages/cloud-cognitive/src/components/Datagrid/_storybook-styles.scss:38]: Can we clarify what this calc is for? Also we typically use `spacing` or `rem` instead of pixel values.

File: packages/cloud-cognitive/src/components/Datagrid/styles/datagrid.scss
Comment (elycheea) [inline packages/cloud-cognitive/src/components/Datagrid/styles/datagrid.scss:269]: Again, think it’d be good to include a comment, but also we should use spacing tokens or rem here.

**Review by elycheea — COMMENTED** [review]
For the values we’re hardcoding in calc, I think it’d be good to clarify why it’s necessary. I know much of the datagrid is already using pixels, but we should use tokens or rems for any of the changes we are making.

No need to go and update all pixel values though, we can address those in a larger release review.

**Comment by elycheea** [comment]
Also just noticed that the data table header now blends into the table. Might have been a conflict when you merged from `main`?

image

File: packages/cloud-cognitive/src/components/Datagrid/_storybook-styles.scss
Comment (Ratheeshrajan) [reply packages/cloud-cognitive/src/components/Datagrid/_storybook-styles.scss:38]: This is only for storybook display. There is information listed under the table about the hidden column. I'll convert the px values into rem.
Screenshot 2022-06-24 at 3 29 22 PM

File: packages/cloud-cognitive/src/components/Datagrid/styles/datagrid.scss
Comment (Ratheeshrajan) [reply packages/cloud-cognitive/src/components/Datagrid/styles/datagrid.scss:269]: Ths calc is for creating room for pagination under the table body. I'll include comments and make it rem value.

**Comment by davidmenendez** [comment]
@elycheea can you please review @Ratheeshrajan response?

**Comment by elycheea** [comment]
@davidmenendez We discussed via Slack. Ratheesh will add the dense header variant and will follow up with design on clarification since the styles are coming from the Carbon default.

**Comment by Ratheeshrajan** [comment]
@elycheea @matthewgallo For dense header, I added one data table header variant. I'm also waiting for Marion's response on the grey10 header background.

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/Datagrid.js
Comment (elycheea) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/Datagrid.js:46]: Since the dense header is more of a variant, it may make more sense to create the variant as a boolean prop rather than another set of DatagridActions.

**Review by elycheea — CHANGES_REQUESTED** [review]
Visually, I think the changes look good.

Upon closer inspection and some discussion with Matt, I think rather than using `DatagridDenseActions`, we should use a boolean prop `dense` or `useDenseHeader` (something that matches our or Carbon’s existing conventions). From there, we can modify the markup as needed to achieve the dense header styles while using the same `DatagridActions`.

This should also be easier for the consumer to switch between dense and non-dense header styles. Thanks for your patience, Ratheesh, and I hope this makes sense.

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/Datagrid.js
Comment (Ratheeshrajan) [reply packages/cloud-cognitive/src/components/Datagrid/Datagrid/Datagrid.js:46]: @elycheea I'll make the changes requested and get back.

Comment (Ratheeshrajan) [reply packages/cloud-cognitive/src/components/Datagrid/Datagrid/Datagrid.js]: @elycheea the consumer can now configure the dense header by setting the property in 'datagridState' as follows,

`const datagridState = useDatagrid({
    columns,
    data,
    gridTitle,
    gridDescription,
    initialState: {
      pageSize: 10,
      pageSizes: [5, 10, 25, 50],
    },
    useDenseHeader: true,
    DatagridActions,
    DatagridPagination,
  });`

**Review by elycheea — DISMISSED** [review]

**Comment by matthewgallo** [comment]
Discussed with both @elycheea and @Ratheeshrajan over slack but there seems to be some layout issues with the pagination appearing at the bottom of the page in the deploy preview. After that is fixed, I think we'll be ready to merge this one!

**Review by elycheea — APPROVED** [review]
Changes look good to me but @matthewgallo may need to take another look so he can approve or dismiss his previous review.

Thanks for coordinating and communicating with us all @Ratheeshrajan! It’s not easy to do with all our timezone differences!

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**
